packages/remix-dev/vite/plugin.ts

```diff
@@ -1336,3 +1368,41 @@ async function getRouteMetadata(
   };
   return info;
 }
+
+async function handleSpaMode(
+  serverBuildPath: string,
+  assetsBuildDirectory: string,
+  mode: string
+) {
+  // Create a handler and call it for the `/` path - rendering down to the
+  // proper HydrateFallback ... or not!  Maybe they have a static landing page
+  // generated from routes/_index.tsx.
+  let build = await import(serverBuildPath);
+  let { createRequestHandler: createNodeRequestHandler } = await import(
+    "@remix-run/node"
+  );
+  let handler = createNodeRequestHandler(build, mode);
+  let response = await handler(new Request("http://localhost/"));
```

Comment:
Please consider the suggestion documented at https://github.com/remix-run/remix/discussions/7638#discussioncomment-7918072 to allow either an index.html or index.js to be generated. It's the smallest change which allows the generation of a Remix App library file that I could think of but maybe the team has better ideas.

Reply:
Still pondering this, but we're going to get this out as an `unstable_ssr` flag without this functionality to start.  Hopefully that'll let some folks play around with it and we can gather some feedback and continue to evaluate this use case.